### PR TITLE
[MM-39796][MM-39872][MM-T1659][MM-T2827] E2E Tests for Popups

### LINKED ISSUE #1910: Desktop E2E Test: MM-T1659 Prevent browser back/forward from firing in OAuth windows
This ticket is to create a test script to automate some testing for the Desktop App.

The testing framework makes use of a few frameworks to successfully automate Desktop App tasks:

  * Playwright: An E2E testing framework provided by Microsoft that currently has “experimental” support for Electron. Using Playwright, you can access individual `BrowserWindows` or `BrowserViews` and automate most functionality provided within those windows. You can also get access to the main Electron process and verify/check functionality in the main process. See here for docs: [https://playwright.dev/docs/api/class-electron](https://playwright.dev/docs/api/class-electron) 
  * RobotJS: A Node.JS Desktop Automation tool that allows us to arbitrary emulate mouse and keyboard inputs. This is used for automation around Electron keyboard shortcuts and the menu bar. It is useful to cover cases that aren’t automatable directly through Playwright (since the menu is not an HTML control). See here for docs: [http://robotjs.io/docs/syntax](http://robotjs.io/docs/syntax)

The test should be written in the file `e2e/specs/popup_test.js`
If the file doesn’t exist, feel free to create it.

If you’re looking for examples, see these PRs:
[https://github.com/mattermost/desktop/pull/1841](https://github.com/mattermost/desktop/pull/1841)
[https://github.com/mattermost/desktop/pull/1843](https://github.com/mattermost/desktop/pull/1843)
[https://github.com/mattermost/desktop/pull/1846](https://github.com/mattermost/desktop/pull/1846)

The test script for this ticket is as follows:

  1. Ensure the test config is using the Demo Mattermost server, and make sure the app opens on said server.
  1. Login to the demo server
  1. In the message box, type `/github connect`
  1. Click the link to open the GitHub OAuth window
  1. With the window focus, using the menu or a keyboard shortcut, go to History > Back

Expected:
User should not be able to navigate back/forward in history on the OAuth window.
The app should not crash or throw any JS errors.

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-39782

### LINKED ISSUE #1915: Desktop E2E Test: MM-T2827 Copy and Paste when connecting Jira plugin
This ticket is to create a test script to automate some testing for the Desktop App.

The testing framework makes use of a few frameworks to successfully automate Desktop App tasks:

  * Playwright: An E2E testing framework provided by Microsoft that currently has “experimental” support for Electron. Using Playwright, you can access individual `BrowserWindows` or `BrowserViews` and automate most functionality provided within those windows. You can also get access to the main Electron process and verify/check functionality in the main process. See here for docs: [https://playwright.dev/docs/api/class-electron](https://playwright.dev/docs/api/class-electron) 
  * RobotJS: A Node.JS Desktop Automation tool that allows us to arbitrary emulate mouse and keyboard inputs. This is used for automation around Electron keyboard shortcuts and the menu bar. It is useful to cover cases that aren’t automatable directly through Playwright (since the menu is not an HTML control). See here for docs: [http://robotjs.io/docs/syntax](http://robotjs.io/docs/syntax)

The test should be written in the file `e2e/specs/popup_test.js`
If the file doesn’t exist, feel free to create it.

If you’re looking for examples, see these PRs:
[https://github.com/mattermost/desktop/pull/1841](https://github.com/mattermost/desktop/pull/1841)
[https://github.com/mattermost/desktop/pull/1843](https://github.com/mattermost/desktop/pull/1843)
[https://github.com/mattermost/desktop/pull/1846](https://github.com/mattermost/desktop/pull/1846)

The test script for this ticket is as follows:

  1. Ensure the test config is using the Demo Mattermost server, and make sure the app opens on said server.
  1. Login to the demo server
  1. In the message box, type `/jira connect`. Verify that a new window opens.
  1. Type something in the username box.
  1. Use Ctrl<ins></ins>C and then Ctrl<ins></ins>A and then Ctrl<ins></ins>V. Verify that the text reappears.

Expected:
See above.

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-39796

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
E2E tests for popup windows:
- Cut/Copy/Paste/Select All in the window
- Stop the window from going Back/Forward in history

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-39796
https://mattermost.atlassian.net/browse/MM-39782
Closes #1915
Closes #1910

```release-note
NONE
```

### CODE REVIEW

**Review by mickmister — APPROVED** [review]
LGTM

File: e2e/specs/popup.test.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    this.timeout(40000);
+
+    const config = env.demoMattermostConfig;
+
+    beforeEach(async () => {
+        env.cleanDataDir();
+        env.createTestUserDataDir();
+        env.cleanTestConfig();
+        fs.writeFileSync(env.configFilePath, JSON.stringify(config));
+        await asyncSleep(1000);
```
Comment (streamer45) [inline e2e/specs/popup.test.js:22]: Out of curiosity what is this sleep for?

**Review by streamer45 — APPROVED** [review]
Nice :+1:

File: e2e/specs/popup.test.js (hunk 1 above)
Comment (devinbinnie) [reply e2e/specs/popup.test.js:22]: Just to make sure that the file has been written before we attempt to start the app. I know it's done synchronously but I ran into situations in the past where the config wasn't there, so it's just to make sure it is.

**Pull request merged**